functions/analytics-svc/src/dao/DBDAO.ts

```diff
@@ -251,15 +251,20 @@ export class DBDAO {
                 ) {
                     delete credentials.user;
                     delete credentials.password;
-                    if (request.headers["x-idp-authorization"]) {
-                        credentials["token"] = JSON.stringify(
+                    if (request.headers["authorization"]) {
+                        const thirdPartyToken =
                             decode(
-                                request.headers["x-idp-authorization"].replace(
-                                    /bearer /i,
-                                    ""
-                                )
-                            )
-                        );
+                                request.headers["authorization"].replace(/bearer /i,"")
+                            )["entraToken"]
+
+                        if(!decode(thirdPartyToken)) {
```

Comment:
```suggestion
                        if(thirdPartyToken && !decode(thirdPartyToken)) {
```

Should it be like this?

Reply:
i was not worried about this case, but more because I didn't notice we will only be supporting with the hana jwt in
```
credentials.dialect === config.DB.HANA &&
                    env.USE_HANA_JWT_AUTHC === "true"
``` 
I thought it should pass through and get to
```
resolve(client);
```
which would be for a non jwt hana i am guessing.

Anyways thanks for the explanation. looks good.